docs/proposals/webhook-validation.md

```diff
@@ -0,0 +1,130 @@
+<!--
+Copyright The Shipwright Contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+---
+title: webhook-validation
+authors:
+  - "@ImJasonH"
+reviewers:
+  - "@gmontero"
+  - "@zhangtbj"
+approvers:
+  - "@qu1queee"
+  - "@adamkaplan"
+creation-date: 2020-03-19
+last-updated: 2020-03-19
+status: provisional
+---
+
+# Webhook Validation
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Open Questions [optional]
+
+Do we want to require that operators run another job to synchronously validate client requests?
+
+Do we want to rely on Knative's webhook packages, or write our own code to manage certs ourselves?
+Is there anything already out there in controller-gen we should be using?
```

Comment:
not a req for change, but rather an aside for those who might look at this, @coreydaley on our team has a non-shipwright assignment that happens to involve investigating the particulars here around webhook cert management outside the confines of knative

hopefully we can take what he discovers and apply it to the consideration here

Reply:
UPDATE:  more details are in https://github.com/shipwright-io/build/issues/596 though some of them are in Red Hat only slack discussion threads that need to be summarized in something consumable by the shipwright community.

High level summary:  there is not at the moment in the operator framework / controller-gen space an equivalent of the auto cert generation / mgmt features in the knavite webhook abstraction layers.